lib/parsers/m3u_parser.rb

```diff
@@ -0,0 +1,25 @@
+class FormatParser::M3UParser
+  include FormatParser::IOUtils
+
+  HEADER = '#EXTM3U'
+
+  def likely_match?(filename)
+    filename =~ /\.(m3u|m3u8)$/i
+  end
+
+  def call(io)
+    io = FormatParser::IOConstraint.new(io)
+
+    header = safe_read(io, 7)
+    return unless HEADER.eql?(header)
+
+    file_content = io.read(io.size)
```

Comment:
Not sure if we should read the whole content, as the file may be huge and it would defy the purpose of reading as little as possible from the file. Returning the format and size should suffice for now.

Reply:
Right! I will remove this part. Looking at the directives here https://en.wikipedia.org/wiki/M3U, do you think there is any more information we should be parsing?

Reply:
Right now I can't think of anything, no 🤔